pennylane/templates/subroutines/fable.py

```diff
@@ -166,7 +166,11 @@ def compute_decomposition(input_matrix, wires, tol=0):  # pylint:disable=argumen
                 for c_wire in nots:
                     op_list.append(qml.CNOT(wires=[c_wire] + ancilla))
                 op_list.append(qml.RY(2 * theta, wires=ancilla))
-                nots[wire_map[control_index]] = 1
+                nots = {}
+                if wire_map[control_index] in nots:
+                    del nots[wire_map[control_index]]
+                else:
+                    nots[wire_map[control_index]] = 1
```

Comment:
```suggestion
                nots[wire_map[control_index]] = 1
```
The code inside the if block shouldn't run ever since we only delete cnots when we have approximated some RY gates away.  The fix I think is actually just adding a `nots = {}` line.